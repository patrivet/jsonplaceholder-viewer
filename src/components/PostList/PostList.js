// Custom components
import Post from "../Post";

const PostList = ({ posts }) => {
  return (
    <div>
      {posts?.map((post) => {
        return <Post key={post.id} post={post} />;
      })}
    </div>
  );
};

export default PostList;
