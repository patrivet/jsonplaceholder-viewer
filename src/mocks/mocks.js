export const mockPosts = [
  { id: 1, title: "title1", body: "a b c d e far walk soon", userId: "1" },
  { id: 2, title: "title2", body: "a b c d e raf klaw noos", userId: "1" },
];
