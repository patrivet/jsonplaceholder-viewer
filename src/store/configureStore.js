import { createStore, compose, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import logger from "redux-logger";
import rootReducer from "./index";

export default function configureStore(initialState) {
  return createStore(
    rootReducer,
    initialState,
    compose(applyMiddleware(thunk, logger))
  );
}
