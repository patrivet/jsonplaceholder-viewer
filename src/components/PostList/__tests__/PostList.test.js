import { render } from "@testing-library/react";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import PostList from "../PostList";
import { mockPosts } from "../../../mocks/mocks";
describe("PostList", () => {
  const mockStore = configureStore();
  const store = mockStore({
    posts: [],
    loading: false,
  });
  // let component;
  test("renders correctly", () => {
    const { container } = render(
      <Provider store={store}>
        <PostList posts={mockPosts} />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
