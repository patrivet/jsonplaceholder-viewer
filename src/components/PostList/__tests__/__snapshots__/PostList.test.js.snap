// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostList renders correctly 1`] = `
<div>
  <div
    class="postList"
  >
    <div
      class="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
        >
          User: 
          1
        </p>
        <p
          class="MuiTypography-root makeStyles-title-2 MuiTypography-body2"
        >
          title1
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          a b c d e far walk soon
        </p>
      </div>
      <div
        class="MuiCardActions-root MuiCardActions-spacing"
      >
        <button
          aria-label="delete"
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
        >
          User: 
          1
        </p>
        <p
          class="MuiTypography-root makeStyles-title-2 MuiTypography-body2"
        >
          title2
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          a b c d e raf klaw noos
        </p>
      </div>
      <div
        class="MuiCardActions-root MuiCardActions-spacing"
      >
        <button
          aria-label="delete"
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
